src/main/java/io/jenkins/plugins/checks/ContextResolver.java

```diff
@@ -22,8 +22,7 @@ public String resolveHeadSha(final SCMSource source, final Run<?, ?> run) {
         try {
             return resolveHeadSha(source.fetch(head, null));
         } catch (IOException | InterruptedException e) {
-            throw new IllegalStateException(String.format("Could not resolve head sha, source: %s, run: %s",
-                    source, run));
+            throw new IllegalStateException("Could not resolve head sha, source: " + e);
```

Comment:
Is there a reason you aren't logging the exception? in the above code you're calling toString implicitly on it and getting the message
```suggestion
            throw new IllegalStateException("Could not resolve head sha, source: ", e);
```

Reply:
No... I got something useful from the e's message just now and so eagerly want to print it that I forget to keep the whole original exception.